File: modules/schema/data_parser.py
from pydantic import BaseModel
from typing import Optional, Dict, Any

from modules.enums import SekaiServerRegion, SekaiEventType, SekaiEventStatus


class WorldBloomChapterStatus(BaseModel):
    server: SekaiServerRegion
    event_id: int
    character_id: int
    chapter_status: SekaiEventStatus


class EventStatus(BaseModel):
    server: SekaiServerRegion
    event_id: int
    event_type: SekaiEventType
    event_status: SekaiEventStatus
    remain: str
    assetbundle_name: str
    chapter_statuses: Optional[Dict[int, WorldBloomChapterStatus]] = None
    detail: Dict[str, Any]

File: utils/model/data_parser.go
package model

// WorldBloomChapterStatus represents world bloom chapter status
type WorldBloomChapterStatus struct {
	Server        SekaiServerRegion `json:"server"`
	EventID       int               `json:"event_id"`
	CharacterID   int               `json:"character_id"`
	ChapterStatus SekaiEventStatus  `json:"chapter_status"`
}

// EventStatus represents event status
type EventStatus struct {
	Server          SekaiServerRegion               `json:"server"`
	EventID         int                             `json:"event_id"`
	EventType       SekaiEventType                  `json:"event_type"`
	EventStatus     SekaiEventStatus                `json:"event_status"`
	Remain          string                          `json:"remain"`
	AssetbundleName string                          `json:"assetbundle_name"`
	ChapterStatuses map[int]WorldBloomChapterStatus `json:"chapter_statuses,omitempty"`
	Detail          map[string]interface{}          `json:"detail"`
}
